app/services/events/create_service.rb

```diff
@@ -77,6 +62,16 @@ def call(organization:, params:, timestamp:, metadata:)
       result
     end
 
+    private
+
+    def customer(organization:, params:)
```

Comment:
I started using e.g. `return @customer if defined? @customer` as a first line so that we also memoize nil values and don't do query each time if customer is not found

Reply:
Good spot 👍